### ISSUE TITLE
Return the error when appcenter cli is not logged in

### ISSUE THREAD

**Issue by jeremybarbet** [getsentry/sentry-cli#1241]
### Environment

How do you use Sentry?
- Saas

Which SDK and version?
- @sentry/cli 2.0.4

### Steps to Reproduce

- Being logged out of `appcennter` (`appcenter logout`)
- Run `sentry-cli react-native appcenter <APP> <PLATFORM> <PATH>`

### Expected Result

When running [this command](https://github.com/getsentry/sentry-cli/blob/master/src/utils/appcenter.rs#L93-L106) directly from the terminal (`appcenter codepush deployment history <DEPLOYMENT> --app <APP> --output json`) that will results to:

```bash
{"succeeded":false,"errorCode":5,"errorMessage":"Command 'appcenter codepush deployment history' requires a logged in user. Use the 'appcenter login' command to log in."}
```

But it's not the case through the rust implementation. I wanted to contribute and return this error, but somehow I got an issue when trying to run the docker command: `docker run --rm -v (pwd):/work sentry-cli react-native appcenter --log-level debug ...` which throw 

```bash
can't execute 'node': no such file or directory
```

In short, would be nice to get this error through the rust cli.

### Actual Result

The command above gives the following error:

```bash
error: Failed to load AppCenter deployment history
  caused by:
```

Created a PR to make the documentation more clear on the docs repository: https://github.com/getsentry/sentry-docs/pull/5063
